### PR TITLE
chore(hybrid-cloud): use sentry performance metrics

### PR DESCRIPTION
**Pull request by corps**
Trying to swap out dd for sentry performance metrics in hybrid cloud code paths.

### CODE REVIEW

**Comment by corps** [comment]
I was hoping that I could use the `start_transaction` as a decorator instead of a context so the diff size would be much much smaller.  I recommend turning off whitespace changes to see whats up.

**Comment by corps** [comment]
Also added automatic metrics for all our hybrid cloud services.  This help us track down performance and make more informed decisions as we expand their usage.  @mikejihbe

**Review by mikejihbe — APPROVED** [review]
I like the auto instrumentation 👍

**Review by markstory — APPROVED** [review]

File: src/sentry/models/outbox.py [hunk 1]
```diff
@@ -171,24 +173,29 @@ def process_coalesced(self) -> Generator[OutboxBase | None, None, None]:
...
-            )
+
+            transaction = sentry_sdk.Hub.current.scope.transaction
+            if transaction:
+                set_measurement(
+                    "outbox.processing_lag",
+                    datetime.datetime.now().timestamp()
+                    - first_coalesced.scheduled_from.timestamp(),
+                    "second",
+                )
```
[the remark this answers is not in the document]
Comment (corps) [reply src/sentry/models/outbox.py:184]: So interestingly, tests fail if I use the `transaction.set_measurement` directly.  I believe that the sentry_sdk may behave differently in tests?  The conditional was necessary in the end.

[the remark this answers is not in the document]
Comment (corps) [reply src/sentry/models/outbox.py:184]: Oh wait yeah I see what you mean, will change.

File: src/sentry/models/outbox.py [hunk 2]
```diff
@@ -156,9 +158,9 @@ def next_schedule(self, now: datetime.datetime) -> datetime.datetime:
         return now + (self.last_delay() * 2)
 
     def save(self, **kwds: Any):
-        tags = {"category": OutboxCategory(self.category).name}
-        metrics.incr("outbox.saved", 1, tags=tags)
-        super().save(**kwds)
+        with sentry_sdk.start_transaction(op="outbox.save", name="save") as transaction:
```
[the remark this answers is not in the document]
Comment (corps) [reply src/sentry/models/outbox.py:161]: Maybe this particular metric then makes most sense as a dd metric.  I'll swap back.

File: src/sentry/services/hybrid_cloud/__init__.py [hunk 3]
```diff
@@ -186,7 +190,58 @@ def silo_mode_delegation(
...
+
+
+def _wrap_with_metrics(
+    m: Callable[..., Any], service_class_name: str, method_name: str
+) -> Callable[..., Any]:
+    def wrapper(*args: Any, **kwds: Any) -> Any:
+        with sentry_sdk.start_transaction(
+            op=f"hybrid_cloud.services.{service_class_name}", name="execute"
+        ):
+            transaction: Transaction | None = Hub.current.scope.transaction
```
[the remark this answers is not in the document]
Comment (corps) [reply src/sentry/services/hybrid_cloud/__init__.py:238]: Interestingly, enough, this caused test failures -- the transaction returned by the context helper was a `NoOp`.  Again, I didn't go very deep to figure out why, but taking the transactional directly from the scope works, while doing it from the context manager doesn't :/

**Pull request merged**